Tomb of Rafid update + new hostile mob (hoppers)

unid15 opened the pull request:
- Tomb of Rafid now has a new puzzle/room: the Water Gallery. It contains a tough puzzle that requires at least two people and extreme coordination to complete.
- Fixed many bugs with the tomb, including rock walls spawning vacuum when destroyed, being able to crowbar water turning it into floors... Also borgs can no longer use ladders :imp:
- Lowered damage done by fire blasts by a tiny amount. Pyromaniacs should no longer instacrit people, but it'll still be very close
- Added hoppers - a family of monsters that resemble frogs in their behavior and (slightly) in their appearance, but don't be fooled - unlike actual frogs they're not going to hunt mosquitoes - they're going to hunt you! The family includes **legionary hoppers** (basic unit), **centurion hoppers** (start with a spear, shorter leap/toss cooldown), **javelineer hoppers** (start with a bag of spears, shorter leap/toss cooldown, keep distance from enemies)

_Hopper Legionaries_
- Legionaries are slow, have 30 health and deal 1-10 damage by punching. They have an ability to leap at their enemies from large distances, quickly closing the gap. If they hit an enemy during leaping, they start holding onto him and beating him with their muscular arms. They don't leap if the enemy is adjacent to them, or if they're holding a spear
- Their punches have a 10% chance to stun IF they're holding on to their victim. They also punch their victim twice as fast if they're holding onto them (effectively doubling their damage output).
- When walking (not holding onto somebody), all hoppers can pick up spears. They can use spears for both melee combat (improving their 1-10 damage to 10-18), and for ranged combat (leap ability is replaced with spear toss ability that throws the spear at enemy, dealing respectable damage)
- Added frog legs (food item). Each hopper corpse can give up to 2 frog legs.
- Hostile infestation event can spawn up to 8 legionary hoppers.

![reeeeeeeeee](https://cloud.githubusercontent.com/assets/9512290/16171988/366dc26c-357e-11e6-925d-f6253936e276.gif)

_Centurion hoppers_
- More powerful version of legionaries, they have a shorter leap/spear throw cooldown and they start with one spear. They also have more health.

_Javelineer hoppers_
- Like centurion frogs, they have a shorter leap/spear throw cooldown, and they start with 4 spears. Unlike other frogs, javelineer hoppers stay away from their enemy, tossing spears at them from a distance. They have slightly more health.

_The Water Gallery_

http://puu.sh/py5qL/5e0c64eecc.png The whole thing (a bit outdated, doesn't include the exit)

http://puu.sh/pxlcz/bd85e06b95.jpg Main puzzle

http://puu.sh/pzzxs/ce67f0ac49.jpg Exit from the water gallery (door can only be opened from inside)

http://puu.sh/pxlcZ/8b137275d8.png Flooded caverns (a bit outdated) (you're dropped off in the tiny room with a toolbox and a flashlight, protected from the frogs by a sandstone wall)
- The puzzle consists of a large body of water that can't be walked on. There are 6 buttons, each of them controlling a group of platforms hidden underwater. Pressing a button raises its group of platforms to the surface, allowing walking on it. There is also a seventh group of platforms, it is lowered and raised every time a button is pressed.
- Only two groups of platforms can be raised at once. Attempting to raise a group of platforms when two are already raised will lower the group which was raised the earliest. For example, raising group C when A and B are already raised will lower A, resulting in B and C being raised. Raising A after that will lower B, resulting in C and A.
- The goal of the puzzle is to find out which platforms belong to which group and then raising platforms needed for your partner to advance. Standing on a lowering platform will teleport you to a large flooded cavern full of **hoppers**, which eventually connects to the tomb's main hall. Most of the water there is shallow and can be walked on
- The reward for completing the puzzle is a telekinesis injector and 2 mind batterers (adminbus only item that affect all humans around you with a 50% chance of stun, each has 2 uses). To get back to the tomb, either ask your partner to press the right buttons again, or dig along the sandstone wall to get to the embalming chamber, from which you can get to the main hallway
- Also telekinesis can now be used to push buttons

![what the fuck is this](https://cloud.githubusercontent.com/assets/9512290/16173286/ca0d1c8e-359c-11e6-9f66-ee2338bdb8d1.gif)

ComicIronic commented on maps/RandomZLevels/tomb.dm:
```diff
@@ -260,3 +263,63 @@
+		else
+			teleport_destination = src
+
+	if(istype(AM, /obj/item) || istype(AM, /obj/machinery) || istype(AM, /obj/structure) || isliving(AM))
```
So mecha stay floating?
Is there any reason for this check?

unid15 replied:
Good catch thanks

It's here to prevent effects and landmarks from drowning. It caused me a lot of headache yesterday when the puzzle stopped working after a few button presses

ComicIronic commented on maps/RandomZLevels/tomb.dm:
```diff
@@ -260,3 +263,63 @@
+
+		return ..()
+
+	else if(last_pressed.len == 2) //Not activated - turning the button on
```
Make this a var on the button so they can be edited to have more than two.

ComicIronic commented on maps/RandomZLevels/tomb.dm:
```diff
@@ -260,3 +263,63 @@
+//SPECIAL BUTTONS
+//Only two can be activated at once
+//Activating a third button when two are activated toggles the first one off
+/obj/structure/button/water_puzzle
```
The type name is not particularly descriptive and these might be used for more than just the water puzzle in the future, so I suggest you rename them.

Shadowmech88 commented:
So what happens if you're standing on a tile, and the other guy changes it to water by pressing a button?
What does standing in water do to you?

AI-is-rogue commented:
@Shadowmech88 

> The goal of the puzzle is to find out which platforms belong to which group and then raising platforms needed for your partner to advance. Standing on a lowering platform will teleport you to a large flooded cavern full of legionnaire frogs, which eventually connects to the tomb's main hall.

unid15 commented:
Renamed legionnaire frogs to better sounding "legionary hoppers"

Hoppers can now pick up and use spears for melee and ranged combat (when they pick up a spear, the leap ability is replaced with spear throw ability, and damage is boosted from 1-10 to 10-18)

Added hopper centurions (shorter ability cooldowns, spawn with a spear) and javelineers (shorter cooldowns, start with multiple spears, stay away from foes instead of running into melee range)

![](http://puu.sh/py4uP/46908a7de9.gif)

unid15 commented:
Fixed many bugs with the tomb and further extended the water gallery. It now has a normal exit (no need to push the buttons again, you'll only have to do some mining)

Also borgs can no longer use ladders. They can still enter the tomb, but they gotta do it legally using the drawbridge

![](http://puu.sh/pzzxs/ce67f0ac49.jpg)

unid15 commented:
Pretty much ready. If adding hoppers to the hostile infestation event is an issue, I'll remove them

Skullyton commented:
I don't know why, but I'm expecting the hoppers to make the same noise as Murlocs

clusterfack commented on code/game/objects/items/weapons/twohanded.dm:
```diff
@@ -83,7 +83,7 @@
 /obj/item/weapon/fireaxe/afterattack(atom/A as mob|obj|turf|area, mob/user as mob, proximity)
 	if(!proximity) return
 	..()
-	if(A && wielded && (istype(A,/obj/structure/window) || istype(A,/obj/structure/grille))) //destroys windows and grilles in one hit
```
???
Why

unid15 replied:
I added it to the grille's attackby(), so there is no feature loss. Did that because otherwise you can destroy invulnerable grilles with fireaxes

clusterfack replied:
Okay

clusterfack commented:
Alright this looks fine to me

Kurfursten commented:
>Put hostile mobs in animal.dmi
REEEEEE FROGS GET OUT
